gluten-celeborn/src/main/java/org/apache/spark/shuffle/gluten/celeborn/CelebornShuffleManager.java

```diff
@@ -240,12 +260,17 @@ public <K, V> ShuffleWriter<K, V> getWriter(
                 celebornConf,
                 h.userIdentifier(),
                 false);
-        if (ShuffleMode.HASH.equals(celebornConf.shuffleWriterMode())) {
+        if (!ShuffleMode.HASH.equals(celebornConf.shuffleWriterMode())) {
+          throw new UnsupportedOperationException(
+              "Unrecognized shuffle write mode!" + celebornConf.shuffleWriterMode());
+        }
+        if (h.dependency() instanceof ColumnarShuffleDependency) {
+          // columnar-based shuffle
           return new CelebornHashBasedColumnarShuffleWriter<>(
               h, context, celebornConf, client, metrics);
         } else {
```

Comment:
ditto.

Reply:
the dependency is awlays instanceof ShuffleDependency.. so it seems the clause is unnecessary